### DISCUSSION TITLE
how to insert an infix as last item in a group

### DISCUSSION THREAD

**Discussion by mooseyboots** [magit/transient#320] [Support] [answered]
i'm using `transient-insert-suffix` to add an infix to an existing prefix (following "Modifying Existing Transients" in the manual, and using the showcase also), and i'm struggling with the LOC argument.

i can use say `(2 -1)`, to identify the last item in group 3, but my new item will be inserted before it. whereas if i choose `(3 0)` it will be inserted as first in the next group. if i use `(3) on its own, it tries to insert the infix as a sibling of the group, which isn't allowed.

is there a way to insert an infix as last item in a group?

(also, i think there is some ambiguity about the LOC arg in the docs, sometimes it says it will be inserted after LOC, sometimes before.)

**Comment by tarsius** [comment, marked as answer]
There are two function for inserting a binding

* `transient-insert-suffix` inserts before LOC.
* `transient-append-suffix` inserts after LOC.

> is there a way to insert an infix as last item in a group?

Use `transient-append-suffix`.

> (also, i think there is some ambiguity about the LOC arg in the docs, sometimes it says it will be inserted after LOC, sometimes before.)

The docstrings of these functions mention the respective correct "direction". In the manual "before" and "after" appear in the shared item about `transient-insert-suffix` and `transient-append-suffix`. The function names and directions appear in the same order, unambiguously linking each direction to just one function, but I guess I could be more verbose.

```diff
 - Function: transient-insert-suffix prefix loc suffix &optional keep-other ::
 - Function: transient-append-suffix prefix loc suffix &optional keep-other ::
 
-  These functions insert the suffix or group {{{var(SUFFIX)}}} into {{{var(PREFIX)}}} before
-  or after {{{var(LOC)}}}.
+  These functions insert the suffix or group {{{var(SUFFIX)}}} into {{{var(PREFIX)}}} next
+  to the location specified by LOC.  ~transient-insert-suffix~ inserts
+  before LOC and ~transient-append-suffix~ inserts after LOC.
```

"before" does not appear a second time in this section. "after" does appear one more time, and it is clear that that instance is about `transient-append-suffix`. So no, there is no ambiguity; you just overlooked that `transient-append-suffix` exists.

**Comment by mooseyboots** [comment]
ah, sorry, that's the trick. problem solved. sorry i missed it in the docs, reading too fast...?
